lib/puppet/parser/catalog_validators/site_validator.rb

```diff
@@ -0,0 +1,22 @@
+module Puppet::Parser
+  module CatalogValidators
+    # Validator that asserts that only application components can appear inside a site.
+  class SiteValidator < CatalogValidator
+
+      def self.stage?(stage)
```

Comment:
the term 'stage' is asily confused with 'puppet stage' - suggest a more descriptive name 'validation_stage', or 'compilation_stage' or some such.